pandas/core/indexes/base.py

```diff
@@ -2110,7 +2110,7 @@ def argsort(self, *args, **kwargs):
         return result.argsort(*args, **kwargs)
 
     def __add__(self, other):
-        return Index(np.array(self) + other)
+        return Index(np.array(self) + other, name=self.name)
```

Comment:
no, this is slightly complicated. the ``__iadd__`` operator needs to be modified (and not ``__add__``), and corresponding ones.